src/api/app/controllers/webui/package_controller.rb

```diff
@@ -991,23 +1000,24 @@ def set_remote_linkinfo
   end
 
   def check_package_name_for_new
-    @package_name = params[:name]
-    @package_title = params[:title]
-    @package_description = params[:description]
+    package_name = params[:package][:name]
 
-    unless Package.valid_name?(@package_name)
-      flash[:error] = "Invalid package name: '#{@package_name}'"
-      redirect_to controller: :project, action: :new_package, project: @project
+    # FIXME: This should be a validation in the Package model
+    unless Package.valid_name?(package_name)
+      flash[:error] = "Invalid package name: '#{package_name}'"
+      redirect_to action: :new, project: @project
       return false
     end
-    if Package.exists_by_project_and_name(@project.name, @package_name)
-      flash[:error] = "Package '#{@package_name}' already exists in project '#{@project}'"
-      redirect_to controller: :project, action: :new_package, project: @project
+    # FIXME: This should be a validation in the Package model
+    if Package.exists_by_project_and_name(@project.name, package_name)
+      flash[:error] = "Package '#{package_name}' already exists in project '#{@project}'"
+      redirect_to action: :new, project: @project
       return false
     end
+    # FIXME: This should be a Pundit policy
     unless User.possibly_nobody.can_create_package_in?(@project)
       flash[:error] = "You can't create packages in #{@project.name}"
-      redirect_to controller: :project, action: :new_package, project: @project
+      redirect_to action: :new, project: @project
```

Comment:
Possible improvement here: because all conditions are basically doing the same thing: updating `flash[:error]`, redirecting to `:new` action we could extract it to a service or method, collection the errors and if error exists, set the error in the flash and redirect...

Reply:
This will all be gone in a following PR to address the FIXMEs.